test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -430,28 +430,45 @@ u256 EVMInstructionInterpreter::eval(
 	return 0;
 }
 
-u256 EVMInstructionInterpreter::evalBuiltin(BuiltinFunctionForEVM const& _fun, const std::vector<u256>& _arguments)
+u256 EVMInstructionInterpreter::evalBuiltin(
+	BuiltinFunctionForEVM const& _fun,
+	vector<Expression> const& _arguments,
+	vector<u256> const& _evaluatedArguments
+)
 {
 	if (_fun.instruction)
-		return eval(*_fun.instruction, _arguments);
-	else if (_fun.name == "datasize"_yulstring)
-		return u256(keccak256(h256(_arguments.at(0)))) & 0xfff;
-	else if (_fun.name == "dataoffset"_yulstring)
-		return u256(keccak256(h256(_arguments.at(0) + 2))) & 0xfff;
-	else if (_fun.name == "datacopy"_yulstring)
+		return eval(*_fun.instruction, _evaluatedArguments);
+
+	string fun = _fun.name.str();
+	if (fun == "datasize" || fun == "dataoffset")
+	{
+		string arg = std::get<Literal>(_arguments.at(0)).value.str();
+		if (arg.length() < 32)
+			arg.resize(32, 0);
+		if (fun == "datasize")
+			return u256(keccak256(arg)) & 0xfff;
```

Comment:
Note to self: This is where the evaluation of the `datasize` instruction in the test case in this PR happens. Earlier literal arguments were evaluated.

Reply:
You can also just add it as a comment to the code.